api/database_engine.go

```diff
@@ -90,6 +94,20 @@ func (e *EverestServer) UpgradeDatabaseEngineOperator(ctx echo.Context, namespac
 		return err
```

Comment:
Wouldn't this always return `nil`?

Reply:
If the operator is at the latest version, it will return nil.

If the operator can be upgraded, i.e, a newer InstallPlan is available, `GetPendingUpgrade` will return that

Reply:
In the context of the dbEngine Status, a pending upgrade refers to a newly available version to upgrade to

Reply:
This PR will provide clarity on this: https://github.com/percona/everest-operator/pull/365

Reply:
Right, but if `u == nil` then we do `return err`. But the `err` comes from `e.kubeClient.GetDatabaseEngine` which is always `nil` at this point.

Reply:
Ah, yes. You're right, didn't notice the typo. Thanks, fixed it.